Automattic/wp-calypso
# Frictionless Email Subscriptions: Add custom signup flow

Related to https://github.com/Automattic/martech/issues/3098

## Proposed Changes

* Adds new `email-subscription` flow to traditional `/start` signup framework
* Adds feature flag for `email-subscription` flow disabled by default
* Adds new `subscribing-email` step to `email-subscription` flow

**To be handled in follow-ups:**
* Error handling
* New user creation fetch request
* Url redirect on account creation
* Tracks events

## Why are these changes being made?


When someone subscribes to an email campaign on a landing page like wordpress.com/learn, we'd like to manage outbound communication PCYsg-6j2-p2. Guides, however, requires a user_id and a WordPress account.

In the past, if the user wanted to subscribe on a landing page and wasn't already logged in, they'd be redirected to the /log-in page. If they didn't have an account, they'd be prompted to create one. Once completed, they'd be redirected to the original landing page.

To polish this experience, we're creating a custom email subscription "signup" flow. For the happy path, the user should be able to subscribe with one click, and the technical implementation should look something like p5uIfZ-f11-p2#comment-22793.

## Testing Instructions



* Check out this branch
* Set the `signup/email-subscription-flow` flag to true in `config/development.json`
* Start local dev
* Navigate to `start/email-subscription` Note that query params are required, so the flow will not be loaded and a blank screen will be shown. ( Handling this error state will be done in a later PR )
* Navigate to `start/email-subscriptions?email=test@gmail.com&redirect_to=wordpress.com/learn&mailing_list=test` Note that the query params are required.
* Verify that a loading screen is initially rendered
* Verify that, shortly afterwards, a passwordless signup form is rendered with the email from the query params autofilled

![2024-06-20 15 59 08](https://github.com/Automattic/wp-calypso/assets/5414230/9d79074e-0048-4e80-9f32-964f439b7122)

## Pre-merge Checklist
